src/main/java/io/vertx/redis/impl/RedisConnection.java

```diff
@@ -248,10 +248,14 @@ void send(final Command<?> command) {
         break;
       case CONNECTING:
       case ERROR:
-        pending.add(command);
+        runOnContext(v -> {
+          pending.add(command);
```

Comment:
my point is that such design create sometimes complex state machines and we can have subtle bugs whereas just using synchronized makes it easier to reason about